src/Illuminate/Routing/Route.php

```diff
@@ -659,9 +666,19 @@ public function setAction(array $action)
      */
     public function gatherMiddleware()
     {
-        return array_unique(array_merge(
+        if (! is_null($this->computedMiddleware)) {
+            return $this->computedMiddleware;
+        }
+
+        // Set incase of an exception, so next time this
+        // method is called, the exception is not rethrown.
+        $this->computedMiddleware = [];
+
+        $this->computedMiddleware = array_unique(array_merge(
```

Comment:
Can't we just directly return it? `return $this->computedMiddleware = array_unique(array_merge(...`